### ISSUE TITLE
Error in basic dependencies with otter.

### ISSUE THREAD

**Issue by abriz97** [quarto-dev/quarto-nvim#86]
Hello, 

I am planning to start using quarto and this plugin looks great. 
However, I am having trouble to get to play with the plugin, and I am getting errors simply by installing the two required plugins
Specifically, I used packer to install the following: 
    use "quarto-dev/quarto-nvim"
    use 'jmbuhr/otter.nvim'
Yet, when opening up a qmd file, i get the following traceback:

> Error executing lua callback: /usr/share/nvim/runtime/filetype.lua:21: Error executing lua: /usr/share/nvim/runtime/filetype.lua:22: Vim(runtime):E5113: Error while calling lua chunk: ...m/site/pack/packer/start
/otter.nvim/lua/otter/keeper.lua:188: attempt to call field 'get_lang' (a nil value)
stack traceback:
        ...m/site/pack/packer/start/otter.nvim/lua/otter/keeper.lua:188: in function 'activate'
        ...m/site/pack/packer/start/quarto-nvim/lua/quarto/init.lua:134: in function 'activate'
        ...m/site/pack/packer/start/quarto-nvim/ftplugin/quarto.lua:20: in main chunk
        [C]: in function 'nvim_cmd'
        /usr/share/nvim/runtime/filetype.lua:22: in function </usr/share/nvim/runtime/filetype.lua:21>
        [C]: in function 'nvim_buf_call'
        /usr/share/nvim/runtime/filetype.lua:21: in function </usr/share/nvim/runtime/filetype.lua:10>
stack traceback:
        [C]: in function 'nvim_cmd'
        /usr/share/nvim/runtime/filetype.lua:22: in function </usr/share/nvim/runtime/filetype.lua:21>
        [C]: in function 'nvim_buf_call'
        /usr/share/nvim/runtime/filetype.lua:21: in function </usr/share/nvim/runtime/filetype.lua:10>
stack traceback:
        [C]: in function 'nvim_buf_call'
        /usr/share/nvim/runtime/filetype.lua:21: in function </usr/share/nvim/runtime/filetype.lua:10>
Error detected while processing BufNewFile Autocommands for "*.qmd"..FileType Autocommands for "*"..function <SNR>1_LoadFTPlugin[19]..script /home/andrea/.local/share/nvim/site/pack/packer/start/quarto-nvim/ftpl
ugin/quarto.lua:
E5113: Error while calling lua chunk: ...m/site/pack/packer/start/otter.nvim/lua/otter/keeper.lua:188: attempt to call field 'get_lang' (a nil value)
stack traceback:
        ...m/site/pack/packer/start/otter.nvim/lua/otter/keeper.lua:188: in function 'activate'
        ...m/site/pack/packer/start/quarto-nvim/lua/quarto/init.lua:134: in function 'activate'
        ...m/site/pack/packer/start/quarto-nvim/ftplugin/quarto.lua:20: in main 
       

I also feel the READMEs of `quarto-nvim` and `quarto-nvim-kickstarter` do not make explicitly clear what the features provided by the plugins are.
Honestly, at the moment I am simply looking for a syntax highlighter, as I am only planning to use R (and I have the great nvim-R installed). 
I think it would also be worth to provide a minimal config that does not require the kick-starter. I am reluctant to just copy and paste files to my config.

Thanks for the plugins though,
Best

**Comment by jmbuhr** [comment]
What part of https://github.com/quarto-dev/quarto-nvim#setup is ambiguous? There is a minimal config mentioned and both example configs are commented. What version of neovim are you on? `vim.treesitter.language.get_lang` (the error you are getting) was only introduced in 0.9, hence the version requirement.

If you only need syntax highlighting you don't need this plugin, but you can steel the treesitter configuration from the example configs.

**Comment by abriz97** [comment]
thanks for the quick reply.

I was using the 0.9-dev version and now the 0.10-dev version, and in both cases the same bug happens.
(there is actually something weirder going on now, whereby if I press "o" or "Enter" in insert-mode, nvim exits, even after un-installing the quarto-nvim and nvim-R package. This is driving me a little crazy, but I don't think this is your plugins fault )

Concerning the documentation, my "issue" is that I would like to know what each plugin is doing.
You mention: "Quarto-nvim provides tools for working on [Quarto](https://quarto.org/) manuscripts in Neovim.", but I feel this should be more specific.

my takeaways from reading the README are:
- autocompletion (but isn't this managed by otter? why do we need quarto-nvim)
- quarto preview (nvim-R has a similar \qp shortcut)
In summary, just by reading the README, it's not clear what the interaction between the features are.
Why do we need otter for a package that provides functions such as `QuartoPreview`? 

Also, there is no mention about other useful stuff such as syntax highlighting.

**Comment by jmbuhr** [comment]
Sounds like something in your setup is broken. Maybe you'll find the root if you build a reproducible example for this or one of the issues you mention.

As mentioned above, syntax highlighting is not provided directly by this plugin, but rather shown in the configuration (that's what treesitter does. Since you have it in your own configuration I assumed you know what it does). For your other questions you may want to follow some of the links in the readme, as I can't cram everything into a single document.
